examples/kubernetes-statefulset/example.go

```diff
@@ -61,5 +61,5 @@ func main() {
 		},
 	}
 	// Now Add some-statefulset.yaml to the config folder.
-	generator.With("config").Add(name+".yaml", encoding.GhodssYAML(set))
+	generator.With("config").WithGeneratedHeader().Add(name+".yaml", encoding.GhodssYAML(set))
```

Comment:
What about chaining `WithTopLevelComment(content string)` and ability to chain that, so in this example, we could do:

`generator := mimic.New()`
`generator = generator.WithTopLevelComment(mimic.GeneratedHeader)`

Reply:
Yup, so I renamed it to `WithTopLevelComment` and made it chainable like `With`. I also added an `Encoder` interface for encoding, which embeds `io.Reader` and a `Commenter` method for commenting actions. 🙂